rllib/policy/tf_policy.py

```diff
@@ -533,6 +533,16 @@ def _build_compute_actions(self,
             raise ValueError(
                 "Must pass in RNN state batches for placeholders {}, got {}".
                 format(self._state_inputs, state_batches))
+
+        # Call the exploration before_forward_pass hook.
+        self.exploration.before_forward_pass(
```

Comment:
You should do this in sampler.py instead of here. Otherwise, we have to change this for each policy.

Reply:
What about TFPolicy.compute_actions? We would have to do it there then as well.

Reply:
Moved into sampler AND TFPolicy.compute_actions.

Reply:
Ah I see.